### PR TITLE
Exposing `diplomat-tool` through a library

### PR DESCRIPTION
**Pull request by robertbastian**
Clients like ICU4X need to version `diplomat-tool` along with the other diplomat crates. Cargo cannot at the moment version binaries, exposing a library is the recommended workaround (https://github.com/rust-lang/cargo/issues/2267).

### CODE REVIEW

**Comment by Manishearth** [comment]
I'd rather not commit to an API here just yet; especially since a thing I definitely want to do is give backends the ability to accept parameters.

**Comment by sffc** [comment]
It's pre-1.0 so personally I don't have a huge concern with making a starter API

File: tool/src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,225 @@
...
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+pub use ast::DocsUrlGenerator;
+
+pub fn gen(
+    entry: &Path,
+    target_language: &str,
```
Comment (Manishearth) [inline tool/src/lib.rs:35]: at the very least I think we should have a config struct or something that encapsulates all this

Comment (Manishearth) [reply tool/src/lib.rs:35]: we should also perhaps have a CLI feature

**Comment by Manishearth** [comment]
Hm, fair. I do think we should spend some effort cleaning up the API though

**Comment by robertbastian** [comment]
There's a lot to clean up in Diplomat. One thing I noticed is that the language gets passed around everywhere as a `&str` instead of an enum, so it's hard to give this a nicer API without bigger changes.

**Review by Manishearth — APPROVED** [review]

**Pull request merged**